src/workerd/io/trace.h

```diff
@@ -708,6 +706,33 @@ struct Span {
         endTime(startTime) {}
 };
 
+// Utility functions for handling span tags.
+void serializeTagValue(rpc::TagValue::Builder builder, const Span::TagValue& value);
+Span::TagValue deserializeTagValue(rpc::TagValue::Reader value);
+
+// Stringify and clone for span tags, getting this to work with KJ_STRINGIFY() appears exceedingly
+// difficult.
+kj::String spanTagStr(const Span::TagValue& tag);
+Span::TagValue spanTagClone(const Span::TagValue& tag);
+
+struct CompleteSpan {
```

Comment:
Not sure if `CompleteSpan` is descriptive enough. Should this be `UserSubmitSpan` or something else? It's unfortunate that we'll need this on top of `Span`, but this needs to store more information than Span has.